Python/ceval.c

```diff
@@ -4674,35 +4705,75 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
 
         TARGET(CALL_NO_KW_TYPE_1) {
             assert(cframe.use_tracing == 0);
-            assert(STACK_ADJUST_IS_RESET);
-            assert(GET_CACHE()->adaptive.original_oparg == 1);
+            DEOPT_IF(call_shape.total_args != 1, CALL);
+            assert(call_shape.kwnames == NULL);
             PyObject *obj = TOP();
             PyObject *callable = SECOND();
-            DEOPT_IF(callable != (PyObject *)&PyType_Type, CALL_NO_KW);
+            DEOPT_IF(callable != (PyObject *)&PyType_Type, CALL);
             PyObject *res = Py_NewRef(Py_TYPE(obj));
-            STACK_SHRINK(1);
             Py_DECREF(callable);
             Py_DECREF(obj);
+            STACK_SHRINK(call_shape.postcall_shrink);
             SET_TOP(res);
             NOTRACE_DISPATCH();
         }
 
-        TARGET(CALL_NO_KW_BUILTIN_CLASS_1) {
+        TARGET(CALL_NO_KW_STR_1) {
             assert(cframe.use_tracing == 0);
-            assert(STACK_ADJUST_IS_RESET);
-            SpecializedCacheEntry *caches = GET_CACHE();
-            _PyAdaptiveEntry *cache0 = &caches[0].adaptive;
-            assert(cache0->original_oparg == 1);
-            PyObject *callable = SECOND();
+            DEOPT_IF(!PyType_Check(call_shape.callable), CALL);
+            PyTypeObject *tp = (PyTypeObject *)call_shape.callable;
+            DEOPT_IF(call_shape.total_args != 1, CALL);
+            DEOPT_IF(tp != &PyUnicode_Type, CALL);
+            STAT_INC(CALL, hit);
+            assert(call_shape.kwnames == NULL);
             PyObject *arg = TOP();
-            DEOPT_IF(!PyType_Check(callable), CALL_NO_KW);
-            PyTypeObject *tp = (PyTypeObject *)callable;
-            DEOPT_IF(tp->tp_version_tag != cache0->version, CALL_NO_KW);
-            STACK_SHRINK(1);
-            PyObject *res = tp->tp_vectorcall((PyObject *)tp, stack_pointer, 1, NULL);
+            PyObject *res = PyObject_Str(arg);
+            Py_DECREF(arg);
+            Py_DECREF(&PyUnicode_Type);
+            STACK_SHRINK(call_shape.postcall_shrink);
             SET_TOP(res);
-            Py_DECREF(tp);
+            if (res == NULL) {
+                goto error;
+            }
+            DISPATCH();
+        }
+
+        TARGET(CALL_NO_KW_TUPLE_1) {
```

Comment:
Might want to note these specializations into the news item too.
``CALL_PY_WITH_DEFAULTS``
``CALL_NO_KW_STR_1``
``CALL_NO_KW_TUPLE_1``
``CALL_BUILTIN_CLASS``
``CALL_BUILTIN_FAST_WITH_KEYWORDS``

Reply:
I'd rather not mention specific specializations in the news, as there are an internal detail and may change at any point.